File: SPRING/igest.server.model/src/main/java/com/yupanastudio/igest/server/principal/dml/Auditoria.java
/*
 * Document   : Auditoria
 * Author     : Franklin Barzola Olivares <frankoantonio@gmail.com>
 */
package com.yupanastudio.igest.server.principal.dml;

import java.time.LocalDate;
import java.time.LocalDateTime;
import org.springframework.data.annotation.Id;

public class Auditoria {

    @Id
    private Long idAuditoria;
    private int anio;
    private String nomEntidad;
    private long idEntidad;
    private String operacion;
    private long idUsuario;
    private String nomUsuario;
    private String ip;
    private String host;
    private String dispositivo;
    private LocalDate feOper;
    private LocalDateTime fhOper;
    private String valores;

    public Long getIdAuditoria() {
        return idAuditoria;
    }

    public void setIdAuditoria(Long idAuditoria) {
        this.idAuditoria = idAuditoria;
    }

    public int getAnio() {
        return anio;
    }

    public void setAnio(int anio) {
        this.anio = anio;
    }

    public String getNomEntidad() {
        return nomEntidad;
    }

    public void setNomEntidad(String nomEntidad) {
        this.nomEntidad = nomEntidad;
    }

    public long getIdEntidad() {
        return idEntidad;
    }

    public void setIdEntidad(long idEntidad) {
        this.idEntidad = idEntidad;
    }

    public String getOperacion() {
        return operacion;
    }

    public void setOperacion(String operacion) {
        this.operacion = operacion;
    }

    public long getIdUsuario() {
        return idUsuario;
    }

    public void setIdUsuario(long idUsuario) {
        this.idUsuario = idUsuario;
    }

    public String getNomUsuario() {
        return nomUsuario;
    }

    public void setNomUsuario(String nomUsuario) {
        this.nomUsuario = nomUsuario;
    }

    public String getIp() {
        return ip;
    }

    public void setIp(String ip) {
        this.ip = ip;
    }

    public String getHost() {
        return host;
    }

    public void setHost(String host) {
        this.host = host;
    }

    public String getDispositivo() {
        return dispositivo;
    }

    public void setDispositivo(String dispositivo) {
        this.dispositivo = dispositivo;
    }

    public LocalDate getFeOper() {
        return feOper;
    }

    public void setFeOper(LocalDate feOper) {
        this.feOper = feOper;
    }

    public LocalDateTime getFhOper() {
        return fhOper;
    }

    public void setFhOper(LocalDateTime fhOper) {
        this.fhOper = fhOper;
    }

    public String getValores() {
        return valores;
    }

    public void setValores(String valores) {
        this.valores = valores;
    }

}

File: FLUTTER/igest_app/lib/models/principal/auditoria.dart
class Auditoria {
  int idAuditoria;
  int anio;
  String nomModulo;
  String nomEntidad;
  int idEntidad;
  String operacion;
  int idUsuario;
  String nomUsuario;
  String ip;
  String host;
  DateTime feOper;
  DateTime fhOper;
  String valores;

  Auditoria({
    required this.idAuditoria,
    required this.anio,
    required this.nomModulo,
    required this.nomEntidad,
    required this.idEntidad,
    required this.operacion,
    required this.idUsuario,
    required this.nomUsuario,
    required this.ip,
    required this.host,
    required this.feOper,
    required this.fhOper,
    required this.valores,
  });

  factory Auditoria.fromJson(Map<String, dynamic> json) {
    return Auditoria(
      idAuditoria: json['id_auditoria'],
      anio: json['anio'],
      nomModulo: json['nom_modulo'],
      nomEntidad: json['nom_entidad'],
      idEntidad: json['id_entidad'],
      operacion: json['operacion'],
      idUsuario: json['id_usuario'],
      nomUsuario: json['nom_usuario'],
      ip: json['ip'],
      host: json['host'],
      feOper: DateTime.parse(json['fe_oper']),
      fhOper: DateTime.parse(json['fh_oper']),
      valores: json['valores'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id_auditoria': idAuditoria,
      'anio': anio,
      'nom_modulo': nomModulo,
      'nom_entidad': nomEntidad,
      'id_entidad': idEntidad,
      'operacion': operacion,
      'id_usuario': idUsuario,
      'nom_usuario': nomUsuario,
      'ip': ip,
      'host': host,
      'fe_oper': feOper.toIso8601String(),
      'fh_oper': fhOper.toIso8601String(),
      'valores': valores,
    };
  }
}
